sphinx/environment.py

```diff
@@ -961,6 +962,35 @@ def process_downloads(self, docname, doctree):
 
     def process_images(self, docname, doctree):
         """Process and rewrite image URIs."""
+        def collect_candidates(imgpath, candidates):
+            globbed = {}
+            for filename in glob(imgpath):
+                new_imgpath = relative_path(path.join(self.srcdir, 'dummy'),
+                                            filename)
+                if filename.lower().endswith('.pdf'):
+                    if 'application/pdf' not in candidates:
+                        globbed.setdefault('application/pdf', []).append(new_imgpath)
+                elif filename.lower().endswith('.svg'):
+                    if 'image/svg+xml' not in candidates:
+                        globbed.setdefault('image/svg+xml', []).append(new_imgpath)
+                else:
+                    try:
+                        f = open(filename, 'rb')
+                        try:
+                            imgtype = imghdr.what(f)
+                        finally:
+                            f.close()
+                    except (OSError, IOError) as err:
+                        self.warn_node('image file %s not readable: %s' %
+                                       (filename, err), node)
+                    if imgtype and 'image/' + imgtype not in candidates:
+                        globbed.setdefault('image/' + imgtype, []).append(new_imgpath)
+            for key, files in iteritems(globbed):
+                if len(files) == 1:
+                    candidates[key] = files[0]
+                else:
+                    candidates[key] = sorted(files, key=len)[0]  # select by similarity
```

Comment:
I think this if-else is not needed. You can just use else clause block even if `len(files) == 1`.